Dip/Dip/Definition.swift

```diff
@@ -50,20 +50,62 @@ func ==(lhs: DefinitionKey, rhs: DefinitionKey) -> Bool {
 public enum ComponentScope {
   /// Indicates that a new instance of the component will be created each time it's resolved.
   case Prototype
+  /// Indicates that instances will be reused during resolve but will be discurded when topmost `resolve` method returns.
+  case ObjectGraph
   /// Indicates that resolved component should be retained by container and always reused.
   case Singleton
 }
 
 ///Definition of type T describes how instances of this type should be created when they are resolved by container.
 public final class DefinitionOf<T>: Definition {
```

Comment:
Ok I see that now, but tbh I'm not sure I like that "nice syntax" of being able to `register { … }.inScope(.ObjectGraph)`. Because having that `inScope` operator being able to change an already-registered dependency means that one can then change it at any time even later after registration, which feels wrong to me:
```
let def = container.register { [unowned container] ClientImp(server: container.resolve() as Server) as Client }
def.inScope(.ObjectGraph)
// … use it
def.inScope(.Singleton)
// … use it again
def.inScope(.Prototype)
// ... use it again
```